### PR TITLE
Propagate NaNs in uniform stats functions

### LINKED ISSUE #38317: jax.scipy.stats.uniform.cdf/pdf/logpdf silently map NaN input to finite values (pdf(nan) == 1.0)
### Description

`jax.scipy.stats.uniform` returns ordinary in-support values for NaN input instead of propagating it:

```python
import jax, numpy as np
import jax.scipy.stats
import scipy.stats

print(jax.scipy.stats.uniform.cdf(np.nan))      # 0.0   (scipy: nan)
print(jax.scipy.stats.uniform.pdf(np.nan))      # 1.0   (scipy: nan)
print(jax.scipy.stats.uniform.logpdf(np.nan))   # -0.0  (scipy: nan)
```

`pdf(nan) = 1.0` is the dangerous one: a NaN entering a likelihood computation is silently treated as a point with full density, so downstream log-likelihoods stay finite and the corruption is undetectable.

For comparison, `torch.distributions.Uniform(0., 1.).cdf(nan)` raises with `validate_args=True` and returns `nan` with `validate_args=False`; scipy returns `nan`.

### Root cause

`uniform.cdf` dispatches with `jnp.select` over comparisons; every comparison with NaN is False, so it falls through to the default `0`:

```python
conds = [lax.lt(x, loc), lax.gt(x, lax.add(loc, scale)), lax.ge(x, loc) & lax.le(x, lax.add(loc, scale))]
vals  = [zero, one, lax.div(lax.sub(x, loc), scale)]
return jnp.select(conds, vals)
```

`pdf`/`logpdf` use the same comparison-mask pattern. Meanwhile `uniform.ppf` in the same file already does the right thing:

```python
return jnp.where(jnp.isnan(q) | (q < 0) | (q > 1), np.nan, ...)
```

The same NaN-masking pattern affects at least `gamma.logpdf(nan, a)` → `-inf` and `poisson.pmf(nan, mu)` → `0.0` (scipy: `nan` for both).

**Suggested fix:** NaN passthrough as in `uniform.ppf`, e.g. a final `jnp.where(jnp.isnan(x), x, result)`. Reproduces in float32 and float64, on 0.10.1 and current `main`.

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.10.1
jaxlib: 0.10.1
numpy:  2.4.6
python: 3.12.13
device: CPU (arm64, macOS)
```

### PR DESCRIPTION
**Pull request by KirtiRamchandani**
## Summary

Fixes #38317.

`jax.scipy.stats.uniform.cdf`, `pdf`, and `logpdf` now propagate NaNs in `x`, `loc`, or `scale` instead of treating NaN comparisons as ordinary in-support values.

## Root cause

- `cdf` uses comparison masks with `jnp.select`; comparisons with NaN are all false, so the default branch returned `0`.
- `logpdf` only checked range comparisons, so NaN inputs fell through to `-log(scale)`, and `pdf` inherited that finite result via `exp(logpdf)`.

## Tests

- Reproduced the pre-fix behavior: `cdf(nan)=0.0`, `pdf(nan)=1.0`, `logpdf(nan)=-0.0` while SciPy returns `nan`.
- `.\.venv\Scripts\python.exe -m pytest tests/scipy_stats_test.py -k testUniformNanPropagation -q`
- `.\.venv\Scripts\python.exe -m pytest tests/scipy_stats_test.py -k "testUniformLogPdf or testUniformCdf or testUniformPpf" -q`
- `.\.venv\Scripts\python.exe -m pytest tests/scipy_stats_test.py -k Uniform -q`
- `.\.venv\Scripts\python.exe -m ruff check jax/_src/scipy/stats/uniform.py tests/scipy_stats_test.py`
- `git diff --check`

### CODE REVIEW

**Comment by KirtiRamchandani** [comment]
Thanks for the review — appreciate the confirmation on NaN propagation behavior for uniform `logpdf`/`cdf`.

Ready to adjust if you'd like explicit scalar/array NaN cases split out.

**Comment by jakevdp** [comment]
This falls under #38651; please note that it may take a while for this to receive attention from the team.
